linode/vpc/framework_models.go

```diff
@@ -19,7 +21,34 @@ type VPCModel struct {
 	Updated     timetypes.RFC3339 `tfsdk:"updated"`
 }
 
-func (m *VPCModel) FlattenVPC(ctx context.Context, vpc *linodego.VPC, preserveKnown bool) {
+type ResourceModel struct {
+	BaseModel
+	IPv6 types.List `tfsdk:"ipv6"`
+}
+
+type ResourceModelIPv6 struct {
+	Range           customtypes.LinodeAutoAllocRangeValue `tfsdk:"range"`
+	AllocationClass types.String                          `tfsdk:"allocation_class"`
+}
+
+var ResourceModelIPv6ObjectType = helper.Must(
+	helper.FrameworkModelToObjectType[ResourceModelIPv6](context.Background()),
+)
+
+type DataSourceModel struct {
+	BaseModel
+	IPv6 types.List `tfsdk:"ipv6"`
+}
+
+type DataSourceModelIPv6 struct {
+	Range customtypes.LinodeAutoAllocRangeValue `tfsdk:"range"`
+}
+
+var DataSourceModelIPv6ObjectType = helper.Must(
+	helper.FrameworkModelToObjectType[DataSourceModelIPv6](context.Background()),
+)
```

Comment:
I had to separate the resource and data source models into two separate structs because of the divergence on the `allocation_class` field, which was causing object/schema mismatch errors otherwise.